src/templates/userSolutionTemplate.tsx

```diff
@@ -4,14 +4,12 @@ import ProblemSolutions from '../components/ProblemSolutions';
 import SubmitProblemSolutionModal from '../components/SubmitProblemSolutionModal';
 import { ProblemInfo } from '../models/problem';
 
-export default function ProblemSolutionsPage(
-  props: PageProps<null, null, { problem: ProblemInfo | null }>
-) {
+export default function Template(props) {
   const [isSubmitModalOpen, setIsSubmitModalOpen] = React.useState(false);
 
-  // Note: In the future, maybe this page should be generated by gatsby
-  // or we can import all the problems as a big JSON file or something.
-  const problem: ProblemInfo = props.location.state?.problem;
+  // problem isn't passed as type of ProblemInfo from gatsby-node,
+  // but all this page's dependencies needs is the problem's name and uniqueId
+  const problem: ProblemInfo = props.pageContext.problem;
```

Comment:
if this is the case, can we change the type from `ProblemInfo` to

```
{ name: string, uniqueId: string }
```

or something?